### PR TITLE
Moved test out of run_all_in_graph_and_eager_mode sparse_image_wrap.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
See #1328

### CODE REVIEW

File: tensorflow_addons/image/sparse_image_warp_test.py [hunk 1]
```diff
@@ -231,42 +231,35 @@ def testSmileyFace(self):
...
-                control_point_locations + control_point_displacements,
-                num_boundary_points=3,
-            )
-            loss = tf.reduce_mean(tf.abs(warped_image - image))
-            return loss
-
-        optimizer = tf.keras.optimizers.SGD(
-            learning_rate=0.001, momentum=0.9, clipnorm=1.0
+def test_that_backprop_runs():
+    """Run optimization to ensure that gradients can be computed."""
```
Comment (seanpmorgan) [inline tensorflow_addons/image/sparse_image_warp_test.py]: Probably need to re-word this doc string now that we're directly checking backprop gradients instead of running an optimization

Comment (gabrieldemarmiesse) [inline tensorflow_addons/image/sparse_image_warp_test.py]:
```suggestion
    """Making sure the gradients can be computed."""
```

Comment (gabrieldemarmiesse) [reply tensorflow_addons/image/sparse_image_warp_test.py]: done
